src/main/java/com/marklogic/client/ext/schemasloader/impl/QbvDocumentFileProcessor.java

```diff
@@ -0,0 +1,150 @@
+
+	final private DatabaseClient databaseClient;
+	final private String qbvGeneratorDatabaseName;
+	final protected List<DocumentFile> qbvFiles = new ArrayList<>();
+	final private XMLDocumentManager docMgr;
+
+
+	/**
+	 * @param databaseClient - a MarkLogic DatabaseClient object for the Schemas database
+	 */
+	public QbvDocumentFileProcessor(DatabaseClient databaseClient, String qbvGeneratorDatabaseName) {
+		this.databaseClient = databaseClient;
+		this.qbvGeneratorDatabaseName = qbvGeneratorDatabaseName;
+		this.docMgr = databaseClient.newXMLDocumentManager();
+	}
+
+	@Override
+	public DocumentFile processDocumentFile(DocumentFile documentFile) {
+		if (isQbvQuery(documentFile)) {
+			// Defer processing until after all other schemas have been processed.
+			qbvFiles.add(documentFile);
+			return null;
+		} else {
+			return documentFile;
+		}
+	}
+
+	private boolean isQbvQuery(DocumentFile documentFile) {
+		String uri = documentFile.getUri();
+		if (uri != null && uri.startsWith("/qbv")) {
+			String extension = documentFile.getFileExtension();
+			if (extension != null) {
+				extension = extension.toLowerCase();
+			}
+			return
+				"sjs".equals(extension)
+					|| "js".equals(extension)
+					|| "xqy".equals(extension)
+					|| "xq".equals(extension);
+		} else return false;
+	}
+
+	public void processQbvFiles() {
+		qbvFiles.forEach(this::processQbvFile);
+		qbvFiles.clear();
+	}
+
+	private void processQbvFile(DocumentFile qbvFile) {
+		ServerEvaluationCall call = getServerEvaluationCall(qbvFile);
+		if (call != null) {
+			StringHandle handleString = new StringHandle();
+			try {
+				call.eval(handleString);
+				if (Format.XML.equals(handleString.getFormat())) {
+					Document xmlDocument = new SAXBuilder().build(new StringReader(handleString.get()));
+					Element root = xmlDocument.getRootElement();
+					if (QBV_XML_ROOT_ELEMENT.equals(root.getName()) & (root.getNamespace() != null && root.getNamespace().getURI().equals(QBV_XML_PLAN_NAMESPACE))) {
+						qbvFile.getDocumentMetadata().getCollections().add(QBV_COLLECTION);
+						String uri = qbvFile.getUri() + ".xml";
+						DOMHandle handleXML = new DOMHandle(new DOMOutputter().output(root.getDocument()));
+						handleXML.setFormat(Format.XML);
+						docMgr.write(uri, qbvFile.getDocumentMetadata(), new JDOMHandle(xmlDocument));
+					} else {
+						throw new RuntimeException(format("Query-Based view generation failed for file: %s; received unexpected response from server: %s", qbvFile.getFile().getAbsolutePath(), handleString.get()));
+					}
+				} else {
+					throw new RuntimeException(format("Query-Based View generation failed for file: %s; ensure your Optic script includes a call to generate a view; received unexpected response from server: %s", qbvFile.getFile().getAbsolutePath(), handleString.get()));
+				}
+			} catch (Exception e) {
+				if ((e instanceof RuntimeException) && (e.getMessage().startsWith("Query-Based View"))) {
```

Comment:
This could cause a null-pointer exception itself - if the original exception is a NullPointerException, the message will be null and then a new NullPointerException will be thrown. 

Given that the only line throwing a checked exception is the `new SAXBuilder` one (this will be true once the DOMOutputter line is removed), just put the try-catch on that line, which removes the outer try-catch. And then you don't need this check here on what kind of exception was thrown.